schemahcl/schemahcl.go

```diff
@@ -479,28 +480,11 @@ func (s *State) mayScopeContext(ctx *hcl.EvalContext, scope []string) *hcl.EvalC
 	if !ok1 && !ok2 {
 		return ctx
 	}
-	nctx := &hcl.EvalContext{
-		Variables: make(map[string]cty.Value, len(vars)),
-		Functions: make(map[string]function.Function, len(funcs)),
-	}
-	for p := ctx; p != nil; p = p.Parent() {
-		for k, v := range p.Variables {
-			if isRef(v) {
-				nctx.Variables[k] = v
-			}
-		}
-	}
-	// Override the parent context with the scoped variables and functions.
-	for n, v := range vars {
-		nctx.Variables[n] = v
-	}
-	for n, f := range funcs {
-		nctx.Functions[n] = f
-	}
-	// A patch from the past. Should be moved
-	// to specific scopes in the future.
-	nctx.Functions["sql"] = rawExprFunc
```

Comment:
Where is this? No longer needed since we don't copy to a new map?

Reply:
global definitions (see stdFuncs).